src/controllers/scripting/javascriptplayerproxy.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include "mixer/basetrackplayer.h"
+#include "track/track.h"
+
+#define PROPERTY_IMPL_GETTER(NAMESPACE, TYPE, NAME, GETTER) \
+    TYPE NAMESPACE::GETTER() const {                        \
+        const TrackPointer pTrack = m_pCurrentTrack;        \
+        if (pTrack == nullptr) {                            \
+            return TYPE();                                  \
+        }                                                   \
+        return pTrack->GETTER();                            \
+    }
+
+class JavascriptPlayerProxy : public QObject {
+    Q_OBJECT
+    Q_PROPERTY(bool isLoaded READ isLoaded NOTIFY trackChanged)
+    Q_PROPERTY(QString artist READ getArtist NOTIFY artistChanged)
+    Q_PROPERTY(QString title READ getTitle NOTIFY titleChanged)
+    Q_PROPERTY(QString album READ getAlbum NOTIFY albumChanged)
+    Q_PROPERTY(QString albumArtist READ getAlbumArtist NOTIFY albumArtistChanged)
+    Q_PROPERTY(QString genre READ getGenre STORED false NOTIFY genreChanged)
+    Q_PROPERTY(QString composer READ getComposer NOTIFY composerChanged)
+    Q_PROPERTY(QString grouping READ getGrouping NOTIFY groupingChanged)
+    Q_PROPERTY(QString year READ getYear NOTIFY yearChanged)
+    Q_PROPERTY(QString trackNumber READ getTrackNumber NOTIFY trackNumberChanged)
+    Q_PROPERTY(QString trackTotal READ getTrackTotal NOTIFY trackTotalChanged)
```

Comment:
Wait, thoses `NOTIFY` don't work as I expected. From JS side, the new value isn't passed to the callback as I thought. You need to write something like:

```js
const player = engine.getPlayer("[Channel1]");
player.trackChanged.connect(() => console.log(`New track: ${player.title}`));
```

Or did I miss something?

Reply:
Ok. I'd modify the code to pass the value but I'll keep it like this if other oppose.